File: src/internal/api/handlers/enterprise.go
package handlers

import (
	"fmt"
	"myobj/src/core/domain/request"
	"myobj/src/core/service"
	"myobj/src/internal/api/middleware"
	"myobj/src/pkg/cache"
	"myobj/src/pkg/logger"
	"myobj/src/pkg/models"

	"github.com/gin-gonic/gin"
)

type EnterpriseHandler struct {
	service     *service.EnterpriseService
	spaceHandler *EnterpriseSpaceHandler
	cache        cache.Cache
}

func NewEnterpriseHandler(service *service.EnterpriseService, spaceHandler *EnterpriseSpaceHandler, cacheLocal cache.Cache) *EnterpriseHandler {
	return &EnterpriseHandler{
		service:      service,
		spaceHandler: spaceHandler,
		cache:        cacheLocal,
	}
}

func (h *EnterpriseHandler) Router(c *gin.RouterGroup) {
	factory := h.service.GetRepository()
	verify := middleware.NewAuthMiddleware(h.cache,
		factory.ApiKey(),
		factory.User(),
		factory.GroupPower(),
		factory.Power())

	enterpriseVerify := middleware.NewEnterpriseMiddleware(
		factory.EnterpriseMember(),
		factory.EnterpriseRole(),
		factory.EnterpriseRolePower(),
		factory.Power())

	enterprise := c.Group("/enterprise")
	enterprise.Use(verify.Verify())
	{
		// 个人操作（不需要企业上下文）
		enterprise.POST("/create", h.CreateEnterprise)
		enterprise.GET("/list", h.GetMyEnterprises)
		enterprise.GET("/info", h.GetEnterpriseInfo)
		enterprise.POST("/switch", h.SwitchEnterprise)
		enterprise.POST("/member/join", h.JoinEnterprise)
		enterprise.POST("/member/leave", h.LeaveEnterprise)
		enterprise.POST("/member/accept", h.AcceptInvite)
		enterprise.POST("/dissolve", h.DissolveEnterprise)

		// 需要企业上下文的操作（成员级别）
		withEnterprise := enterprise.Group("")
		withEnterprise.Use(enterpriseVerify.Verify())
		{
			withEnterprise.GET("/member/list", h.GetMemberList)
			withEnterprise.GET("/role/list", h.GetRoleList)
			withEnterprise.GET("/member/pending", h.GetPendingInvites)

			// 共享空间操作（需要企业上下文 + 具体权限）
			if h.spaceHandler != nil {
				space := withEnterprise.Group("/space")
				{
					space.POST("/mkdir", enterpriseVerify.PowerVerify("enterprise:space:upload"), h.spaceHandler.CreateDir)
					space.GET("/list", enterpriseVerify.PowerVerify("enterprise:space:download"), h.spaceHandler.ListFiles)
					space.POST("/upload/precheck", enterpriseVerify.PowerVerify("enterprise:space:upload"), h.spaceHandler.UploadPrecheck)
					space.POST("/upload", enterpriseVerify.PowerVerify("enterprise:space:upload"), h.spaceHandler.UploadFile)
					space.POST("/delete", enterpriseVerify.PowerVerify("enterprise:space:delete"), h.spaceHandler.DeleteFile)
					space.POST("/delete-dir", enterpriseVerify.PowerVerify("enterprise:space:delete"), h.spaceHandler.DeleteDir)
					space.POST("/rename", enterpriseVerify.PowerVerify("enterprise:space:upload"), h.spaceHandler.RenameFile)
					space.POST("/rename-dir", enterpriseVerify.PowerVerify("enterprise:space:upload"), h.spaceHandler.RenameDir)
					space.GET("/download", enterpriseVerify.PowerVerify("enterprise:space:download"), h.spaceHandler.DownloadFile)
					space.GET("/usage", h.spaceHandler.GetSpaceUsage)
				}
			}

			// 审计日志（需要 enterprise:audit:view 权限）
			auditGroup := withEnterprise.Group("/audit")
			auditGroup.Use(enterpriseVerify.PowerVerify("enterprise:audit:view"))
			{
				auditGroup.GET("/list", h.GetEnterpriseAuditLogs)
				auditGroup.GET("/export", h.ExportEnterpriseAuditLogs)
			}

			// 需要企业管理员权限的操作
			admin := withEnterprise.Group("")
			admin.Use(enterpriseVerify.AdminVerify())
			{
				admin.PUT("/update", h.UpdateEnterprise)
				admin.POST("/member/invite", enterpriseVerify.PowerVerify("enterprise:member:invite"), h.InviteMember)
				admin.GET("/member/invite-code", h.GetInviteCode)
				admin.POST("/member/refresh-code", h.RefreshInviteCode)
				admin.PUT("/member/role", enterpriseVerify.PowerVerify("enterprise:role:manage"), h.UpdateMemberRole)
				admin.POST("/member/remove", enterpriseVerify.PowerVerify("enterprise:member:remove"), h.RemoveMember)
				admin.POST("/role/create", enterpriseVerify.PowerVerify("enterprise:role:manage"), h.CreateRole)
				admin.PUT("/role/update", enterpriseVerify.PowerVerify("enterprise:role:manage"), h.UpdateRole)
				admin.DELETE("/role/delete", enterpriseVerify.PowerVerify("enterprise:role:manage"), h.DeleteRole)
				admin.GET("/powers", h.GetAllPowers)
				admin.POST("/transfer", h.TransferOwnership)
				admin.POST("/toggle-state", h.ToggleEnterpriseState)
				admin.POST("/space/set-quota", h.SetEnterpriseQuota)
			}
		}
	}

	logger.LOG.Info("[路由] 企业路由注册完成✔️")
}

func (h *EnterpriseHandler) CreateEnterprise(c *gin.Context) {
	req := new(request.CreateEnterpriseRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.CreateEnterprise(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) GetMyEnterprises(c *gin.Context) {
	result, err := h.service.GetMyEnterprises(c.GetString("userID"))
	if err != nil {
		c.JSON(200, models.NewJsonResponse(500, "查询失败", err.Error()))
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) GetEnterpriseInfo(c *gin.Context) {
	enterpriseID := c.Query("enterprise_id")
	if enterpriseID == "" {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", nil))
		return
	}
	result, err := h.service.GetEnterpriseInfo(enterpriseID, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) UpdateEnterprise(c *gin.Context) {
	req := new(request.UpdateEnterpriseRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.UpdateEnterprise(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) SwitchEnterprise(c *gin.Context) {
	req := new(request.SwitchEnterpriseRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.SwitchEnterprise(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) InviteMember(c *gin.Context) {
	req := new(request.InviteMemberRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.InviteMember(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) GetInviteCode(c *gin.Context) {
	enterpriseID := c.Query("enterprise_id")
	if enterpriseID == "" {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", nil))
		return
	}
	result, err := h.service.GetInviteCode(enterpriseID, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) RefreshInviteCode(c *gin.Context) {
	enterpriseID := c.Query("enterprise_id")
	if enterpriseID == "" {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", nil))
		return
	}
	result, err := h.service.RefreshInviteCode(enterpriseID, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) JoinEnterprise(c *gin.Context) {
	req := new(request.JoinEnterpriseRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.JoinEnterprise(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) GetMemberList(c *gin.Context) {
	req := new(request.EnterpriseListRequest)
	if err := c.ShouldBindQuery(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.GetMemberList(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) UpdateMemberRole(c *gin.Context) {
	req := new(request.UpdateMemberRoleRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.UpdateMemberRole(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) RemoveMember(c *gin.Context) {
	req := new(request.RemoveMemberRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.RemoveMember(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) LeaveEnterprise(c *gin.Context) {
	req := new(request.LeaveEnterpriseRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.LeaveEnterprise(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) GetPendingInvites(c *gin.Context) {
	result, err := h.service.GetPendingInvites(c.GetString("userID"))
	if err != nil {
		c.JSON(200, models.NewJsonResponse(500, "查询失败", err.Error()))
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) AcceptInvite(c *gin.Context) {
	inviteID := c.Query("invite_id")
	if inviteID == "" {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", nil))
		return
	}
	result, err := h.service.AcceptInvite(inviteID, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) GetRoleList(c *gin.Context) {
	enterpriseID := c.Query("enterprise_id")
	if enterpriseID == "" {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", nil))
		return
	}
	result, err := h.service.GetRoleList(enterpriseID, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) CreateRole(c *gin.Context) {
	req := new(request.CreateEnterpriseRoleRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.CreateRole(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) UpdateRole(c *gin.Context) {
	req := new(request.UpdateEnterpriseRoleRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.UpdateRole(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) DeleteRole(c *gin.Context) {
	req := new(request.DeleteEnterpriseRoleRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.DeleteRole(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) GetAllPowers(c *gin.Context) {
	result, err := h.service.GetAllPowers()
	if err != nil {
		c.JSON(200, models.NewJsonResponse(500, "查询失败", err.Error()))
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) TransferOwnership(c *gin.Context) {
	req := new(request.TransferOwnershipRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.TransferOwnership(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) DissolveEnterprise(c *gin.Context) {
	req := new(request.DissolveEnterpriseRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.DissolveEnterprise(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) ToggleEnterpriseState(c *gin.Context) {
	req := new(request.ToggleEnterpriseStateRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.ToggleEnterpriseState(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) SetEnterpriseQuota(c *gin.Context) {
	req := new(request.SetEnterpriseQuotaRequest)
	if err := c.ShouldBindJSON(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.SetEnterpriseQuota(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) GetEnterpriseAuditLogs(c *gin.Context) {
	req := new(request.EnterpriseAuditListRequest)
	if err := c.ShouldBindQuery(req); err != nil {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", err.Error()))
		return
	}
	result, err := h.service.GetEnterpriseAuditLogs(req, c.GetString("userID"))
	if err != nil {
		c.JSON(200, result)
		return
	}
	c.JSON(200, result)
}

func (h *EnterpriseHandler) ExportEnterpriseAuditLogs(c *gin.Context) {
	enterpriseID := c.Query("enterprise_id")
	if enterpriseID == "" {
		c.JSON(200, models.NewJsonResponse(400, "参数错误", nil))
		return
	}

	csvData, err := h.service.ExportEnterpriseAuditLogs(
		enterpriseID,
		c.Query("action"),
		c.Query("keyword"),
		c.Query("start_time"),
		c.Query("end_time"),
		c.GetString("userID"),
	)
	if err != nil {
		c.JSON(200, models.NewJsonResponse(500, "导出失败", err.Error()))
		return
	}

	c.Header("Content-Type", "text/csv; charset=utf-8")
	eid := enterpriseID
	if len(eid) > 8 {
		eid = eid[:8]
	}
	c.Header("Content-Disposition", fmt.Sprintf("attachment; filename=enterprise_audit_%s.csv", eid))
	c.Data(200, "text/csv; charset=utf-8", csvData)
}

File: webview/packages/api/enterprise.ts
import { get, post, put, del, download } from '@myobj/http'
import { filterParams, API_BASE_URL, API_ENDPOINTS } from '@myobj/shared'
import type {
  ApiResponse,
  Enterprise,
  EnterpriseListResponse,
  EnterpriseMemberListResponse,
  EnterpriseRoleListResponse,
  EnterpriseAuditListResponse,
  SharedFileListResponse,
  SpaceUsage,
  EnterprisePower,
  InviteCodeResponse,
  PendingInviteResponse,
  CreateEnterpriseRequest,
  UpdateEnterpriseRequest,
  SwitchEnterpriseRequest,
  InviteMemberRequest,
  JoinEnterpriseRequest,
  UpdateMemberRoleRequest,
  RemoveMemberRequest,
  LeaveEnterpriseRequest,
  CreateRoleRequest,
  UpdateRoleRequest,
  DeleteRoleRequest,
  TransferOwnershipRequest,
  DissolveEnterpriseRequest,
  ToggleEnterpriseStateRequest,
  SetEnterpriseQuotaRequest,
  EnterpriseListRequest,
  EnterpriseAuditListRequest,
  SharedFileListRequest,
  CreateSharedDirRequest,
  DeleteSharedFileRequest,
  SharedUploadPrecheckRequest
} from '@myobj/shared'

// ========== 企业管理 API ==========

export const createEnterprise = (data: CreateEnterpriseRequest) => {
  return post<ApiResponse<{ enterprise_id: string; invite_code: string }>>(API_ENDPOINTS.ENTERPRISE.CREATE, data)
}

export const getMyEnterprises = () => {
  return get<ApiResponse<EnterpriseListResponse>>(API_ENDPOINTS.ENTERPRISE.LIST)
}

export const getEnterpriseInfo = (enterprise_id: string) => {
  return get<ApiResponse<Enterprise>>(API_ENDPOINTS.ENTERPRISE.INFO, { enterprise_id })
}

export const updateEnterprise = (data: UpdateEnterpriseRequest) => {
  return put<ApiResponse>(API_ENDPOINTS.ENTERPRISE.UPDATE, data)
}

export const switchEnterprise = (data: SwitchEnterpriseRequest) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SWITCH, data)
}

export const transferOwnership = (data: TransferOwnershipRequest) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.TRANSFER, data)
}

export const dissolveEnterprise = (data: DissolveEnterpriseRequest) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.DISSOLVE, data)
}

export const toggleEnterpriseState = (data: ToggleEnterpriseStateRequest) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.TOGGLE_STATE, data)
}

export const setEnterpriseQuota = (data: SetEnterpriseQuotaRequest) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SET_QUOTA, data)
}

// ========== 成员管理 API ==========

export const getMemberList = (params: EnterpriseListRequest) => {
  return get<ApiResponse<EnterpriseMemberListResponse>>(API_ENDPOINTS.ENTERPRISE.MEMBER.LIST, filterParams(params))
}

export const inviteMember = (data: InviteMemberRequest) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.MEMBER.INVITE, data)
}

export const removeMember = (data: RemoveMemberRequest) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.MEMBER.REMOVE, data)
}

export const updateMemberRole = (data: UpdateMemberRoleRequest) => {
  return put<ApiResponse>(API_ENDPOINTS.ENTERPRISE.MEMBER.UPDATE_ROLE, data)
}

export const leaveEnterprise = (data: LeaveEnterpriseRequest) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.MEMBER.LEAVE, data)
}

export const joinEnterprise = (data: JoinEnterpriseRequest) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.MEMBER.JOIN, data)
}

export const acceptInvite = (invite_id: string) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.MEMBER.ACCEPT, {}, { params: { invite_id } })
}

export const getInviteCode = (enterprise_id: string) => {
  return get<ApiResponse<InviteCodeResponse>>(API_ENDPOINTS.ENTERPRISE.MEMBER.INVITE_CODE, { enterprise_id })
}

export const refreshInviteCode = (enterprise_id: string) => {
  return post<ApiResponse<{ invite_code: string }>>(API_ENDPOINTS.ENTERPRISE.MEMBER.REFRESH_CODE, {}, { params: { enterprise_id } })
}

export const getPendingInvites = () => {
  return get<ApiResponse<PendingInviteResponse[]>>(API_ENDPOINTS.ENTERPRISE.MEMBER.PENDING)
}

// ========== 角色管理 API ==========

export const getRoleList = (enterprise_id: string) => {
  return get<ApiResponse<EnterpriseRoleListResponse>>(API_ENDPOINTS.ENTERPRISE.ROLE.LIST, { enterprise_id })
}

export const createRole = (data: CreateRoleRequest) => {
  return post<ApiResponse<{ role_id: string }>>(API_ENDPOINTS.ENTERPRISE.ROLE.CREATE, data)
}

export const updateRole = (data: UpdateRoleRequest) => {
  return put<ApiResponse>(API_ENDPOINTS.ENTERPRISE.ROLE.UPDATE, data)
}

export const deleteRole = (data: DeleteRoleRequest) => {
  return del<ApiResponse>(API_ENDPOINTS.ENTERPRISE.ROLE.DELETE, { data })
}

export const getAllPowers = () => {
  return get<ApiResponse<EnterprisePower[]>>(API_ENDPOINTS.ENTERPRISE.POWERS)
}

// ========== 共享空间 API ==========

export const createSharedDir = (data: CreateSharedDirRequest) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.MKDIR, data)
}

export const getSharedFileList = (params: SharedFileListRequest) => {
  return get<ApiResponse<SharedFileListResponse>>(API_ENDPOINTS.ENTERPRISE.SPACE.LIST, filterParams(params))
}

export const sharedUploadPrecheck = (data: SharedUploadPrecheckRequest) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.UPLOAD_PRECHECK, data)
}

export const deleteSharedFile = (data: DeleteSharedFileRequest, enterprise_id?: string) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.DELETE, data, enterprise_id ? { params: { enterprise_id } } : {})
}

export const downloadSharedFile = (id: string, enterprise_id?: string) => {
  let url = API_ENDPOINTS.ENTERPRISE.SPACE.DOWNLOAD + '?id=' + id
  if (enterprise_id) url += '&enterprise_id=' + enterprise_id
  return download(url, id)
}

export const getSpaceUsage = (enterprise_id: string) => {
  return get<ApiResponse<SpaceUsage>>(API_ENDPOINTS.ENTERPRISE.SPACE.USAGE, { enterprise_id })
}

export const deleteSharedDir = (id: number, enterprise_id?: string) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.DELETE_DIR, { id }, enterprise_id ? { params: { enterprise_id } } : {})
}

export const renameSharedFile = (id: string, name: string, enterprise_id?: string) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.RENAME, { id, name }, enterprise_id ? { params: { enterprise_id } } : {})
}

export const renameSharedDir = (id: number, name: string, enterprise_id?: string) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.RENAME_DIR, { id, name }, enterprise_id ? { params: { enterprise_id } } : {})
}

// 预览（返回 URL）
export const previewSharedFile = (id: string) => {
  return API_BASE_URL + API_ENDPOINTS.ENTERPRISE.SPACE.PREVIEW + '?id=' + id
}

// 缩略图（返回 URL）
export const getSharedFileThumbnail = (fileId: string) => {
  return API_BASE_URL + API_ENDPOINTS.ENTERPRISE.SPACE.THUMBNAIL + '/' + fileId
}

// 搜索文件
export const searchSharedFiles = (params: { enterprise_id: string; keyword: string; page: number; pageSize: number }) => {
  return get<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.SEARCH, filterParams(params))
}

// 获取目录树
export const getSharedPathTree = (enterprise_id: string) => {
  return get<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.PATH_TREE, { enterprise_id })
}

// 移动文件
export const moveSharedFile = (data: { enterprise_id: string; file_id: string; target_path_id: number }) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.MOVE, data)
}

// 创建打包下载任务
export const createPackage = (data: { enterprise_id: string; file_ids: string[]; package_name?: string }) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.PACKAGE, data)
}

// 查询打包进度
export const getPackageProgress = (package_id: string) => {
  return get<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.PACKAGE_PROGRESS, { package_id })
}

// 下载打包文件（返回 URL）
export const downloadPackage = (package_id: string) => {
  return API_ENDPOINTS.ENTERPRISE.SPACE.PACKAGE_DOWNLOAD + '?package_id=' + package_id
}

// 解压冲突检测
export const extractCheck = (data: { enterprise_id: string; file_id: string; target_path_id?: number; file_password?: string }) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.EXTRACT_CHECK, data)
}

// 开始解压
export const extractStart = (data: { enterprise_id: string; file_id: string; target_path_id?: number; file_password?: string; conflict_strategy?: string }) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.EXTRACT, data)
}

// 查询解压进度
export const getExtractProgress = (task_id: string) => {
  return get<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.EXTRACT_PROGRESS, { task_id })
}

// 创建文件分享
export const createShare = (data: { enterprise_id: string; file_id: string; expire?: number; password?: string }) => {
  return post<ApiResponse>(API_ENDPOINTS.ENTERPRISE.SPACE.SHARE, data)
}

// ========== 审计日志 API ==========

export const getEnterpriseAuditLogs = (params: EnterpriseAuditListRequest) => {
  return get<ApiResponse<EnterpriseAuditListResponse>>(API_ENDPOINTS.ENTERPRISE.AUDIT.LIST, filterParams(params))
}

export const exportEnterpriseAuditLogs = (params: Omit<EnterpriseAuditListRequest, 'page' | 'pageSize'>) => {
  const query = new URLSearchParams()
  query.set('enterprise_id', params.enterprise_id)
  if (params.action) query.set('action', params.action)
  if (params.keyword) query.set('keyword', params.keyword)
  if (params.start_time) query.set('start_time', params.start_time)
  if (params.end_time) query.set('end_time', params.end_time)
  const qs = query.toString()
  const url = API_ENDPOINTS.ENTERPRISE.AUDIT.EXPORT + (qs ? '?' + qs : '')
  const filename = `enterprise_audit_${new Date().toISOString().slice(0, 19).replace(/[-:T]/g, '')}.csv`
  return download(url, filename)
}
